scripts/make_s2and_mini_dataset.py

```diff
@@ -0,0 +1,141 @@
+import os
+import json
+import pickle
+import numpy as np
+import pandas as pd
+from s2and.consts import CONFIG
+
+DATA_DIR = CONFIG["main_data_dir"]
+
+OUTPUT_DIR = os.path.join(DATA_DIR, "s2and_mini")
+if not os.path.exists(OUTPUT_DIR):
+    os.mkdir(OUTPUT_DIR)
+
+# excluding MEDLINE because it has no clusters
+DATASETS = [
+    "aminer",
+    "arnetminer",
+    "inspire",
+    "kisti",
+    "pubmed",
+    "qian",
+    "zbmath",
+]
+
+# load all of the artifacts of each dataset
+clusters_all = []
+signatures_all = []
+X_all = []
+keys_all = []
+papers_all = []
+for dataset in DATASETS:
+    print()
+    print(f"Loading data from {dataset}...")
+
+    for file_name in os.listdir(os.path.join(DATA_DIR, dataset)):
+        file_name = os.path.join(DATA_DIR, dataset, file_name)
+        if "specter" in file_name:
+            with open(file_name, "rb") as _pickle_file:
+                X, keys = pickle.load(_pickle_file)
+                X_all.append(X)
+                keys_all.append(keys)
+        elif "cluster" in file_name:
+            with open(file_name) as _json_file:
+                clusters = json.load(_json_file)
+                new_clusters = {}
+                for cluster_id, v in clusters.items():
+                    new_cluster_id = f"{dataset}_{cluster_id}"
+                    new_v = {
+                        "cluster_id": new_cluster_id,
+                        "signature_ids": [f"{dataset}_{i}" for i in v["signature_ids"]],
+                        "model_version": v["model_version"],
+                    }
+                    new_clusters[new_cluster_id] = new_v
+                clusters_all.append(new_clusters)
+        elif "paper" in file_name:
+            with open(file_name) as _json_file:
+                papers = json.load(_json_file)
+                papers_all.append(papers)
+        elif "signature" in file_name:
+            with open(file_name) as _json_file:
+                signatures = json.load(_json_file)
+                new_signatures = {}
+                for signature_id, v in signatures.items():
+                    new_signature_id = f"{dataset}_{signature_id}"
+                    new_v = {
+                        "author_id": v["author_id"],  # maybe this needs to be prepended by dataset?
+                        "paper_id": v["paper_id"],
+                        "signature_id": new_signature_id,
+                        "author_info": v["author_info"],
+                    }
+                    new_signatures[new_signature_id] = new_v
+                signatures_all.append(new_signatures)
+        else:
+            print(f"WARNING: Ignoring {file_name} in {dataset}")
+
+
+# the goal is speed so we'll remove the largest blocks (x > 250 signatures)
```

Comment:
Comment looks out of sync with code; maybe a named constant would be good

Reply:
Will do.